### PR TITLE
Data User Agreement (DUA): backend + frontend DUA file upload

### PR DESCRIPTION
**Pull request by michaelkain**
- contains the full REST interface for DUA management (getDUAsForUser, acceptDUA)
- upload a DUA file to a study + info box
- backend: DUA table managed next to StudyUser, StudyUser extension with isConfirmed
- backend: synchronise user_rights changes to other ms services

missing: more tests, deletion of DUA file (same for protocol file), db_changes scripts

### CODE REVIEW

**Comment by michaelkain** [comment]
Already merged with develop before opening pull request

**Comment by michaelkain** [comment]
@Nunien Hi Julien, the json object of getDUAsForUserId looks the following:
[
    {
        "id": 1,
        "userId": 3,
        "timestampOfNew": 1613053332000,
        "timestampOfAccepted": null,
        "studyId": 1
    }
]

**Comment by michaelkain** [comment]
Great: my regression tests (studies without DUA) still work fine on my PR

**Comment by michaelkain** [comment]
My tests for the new feature work fine too, remaining work: db_change scripts + one bug to discuss with expert - view dataset list, not sure if related to my dev

**Comment by michaelkain** [comment]
[Testing of DUA.pdf](https://github.com/fli-iam/shanoir-ng/files/5982093/Testing.of.DUA.pdf)
my test report

**Comment by michaelkain** [comment]
and I need to finish the migration scripts

**Comment by michaelkain** [comment]
by the way: sorry, that this is not the smallest PR, I will try next time to better split my devs!

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java [hunk 1]
```diff
@@ -223,4 +225,72 @@ void exportBIDSByStudyId(
...
+
+	@ApiOperation(value = "", notes = "Download DUA of a study", tags = {})
+	@ApiResponses(value = {
+	        @ApiResponse(code = 200, message = "dua downloaded", response = Resource.class),
+			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
+			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
+			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
+			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
+	@GetMapping(value = "dua-download/{studyId}/{fileName:.+}/")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_DOWNLOAD'))")
```
Comment (jcomedouteau) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:278]: If a user has the 'can see' right only, he does not have to sign the DUA ?

File: shanoir-ng-studies/src/main/resources/application.yml [hunk 2]
```diff
@@ -131,6 +131,8 @@ ms.url:
   shanoir-ng-studycards: http://localhost:9904/studycards
   shanoir-ng-examinations: http://localhost:9904/examinations
   shanoir-ng-bids: http://localhost:9904/bids
+# data paths
+studies-data: /tmp
```
Comment (jcomedouteau) [inline shanoir-ng-studies/src/main/resources/application.yml:135]: I don't think this should be here

Comment (michaelkain) [reply shanoir-ng-studies/src/main/resources/application.yml:135]: it is in dev profile only, and to better test locally it is fine?

**Comment by jcomedouteau** [comment]
Seems to be OK for me, except for application.yml and a small question !
Missing SQL update too :)

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java (hunk 1 above)
Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:278]: good finding!!! I will change it!

Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:278]: the dua download will require no rights, as I remove all rights during pending state "before acceptance"

File: shanoir-ng-studies/src/main/resources/application.yml (hunk 2 above)
Comment (jcomedouteau) [reply shanoir-ng-studies/src/main/resources/application.yml:135]: Ok, why not

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java (hunk 1 above)
Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java:278]: but the user should see it anyway

Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java]: changed

**Comment by jcomedouteau** [comment]
There is some merge to do as Julien merged my PR, call me if necessary !

**Comment by michaelkain** [comment]
Thank you Jean-Come, I will start the merge and let you know :-)

**Comment by michaelkain** [comment]
Hi Jean-Come, as soon as you give me the go, I will merge it to be deployed tomorrow morning. Cheers, Michael

**Comment by michaelkain** [comment]
by the way, that you may not be suprised: I moved your db_changes file from folder /study to /studies

**Comment by jcomedouteau** [comment]
> by the way, that you may not be suprised: I moved your db_changes file from folder /study to /studies

Yep perfect, we discussed this point with julien too !

**Comment by jcomedouteau** [comment]
For the code, it's good for me. 
Is the front connected yet ? I could not sign the DUA (but I could upload the file and download it)
If yes, it's all good for me, we can merge !

**Comment by Nunien** [comment]
Please don't merge I am connecting it to the other challenge devs, I will submit a new pr that contains this one's devs.

**Comment by michaelkain** [comment]
@Nunien Hi Julien, maybe better accepting this one and then merging with develop?

**Comment by michaelkain** [comment]
that is what I did this morning with the PR of Jean-Come

**Comment by michaelkain** [comment]
Hi, I had Julien on the phone, he will merge this PR first, thank you!
Another good news: I tested more my PR: I could not reproduce the error with view datasets for expert users! So, we should be good.

**Pull request merged**